src/make_watertight/Arc.cpp

```diff
@@ -0,0 +1,713 @@
+            adj_curves[j].merge(temp_curves);
+          }
+        }
+      }
+    }
+
+    // now find the curves that have matching endpts
+    moab::Range candidate_curves;
+    // select curves that do not have coincident front AND back points
+    // place them into candidate curves
+    candidate_curves = intersect(adj_curves[0], adj_curves[1]);
+    if (candidate_curves.empty())
+      continue;
+
+    // subtract the current curve
+    int n_before = candidate_curves.size();
+    candidate_curves.erase(*i);
+    int n_after = candidate_curves.size();
+
+
+    // now find curves who's interior vertices are also coincident and merge them
+    for (moab::Range::iterator j = candidate_curves.begin(); j != candidate_curves.end(); j++) {
+      std::vector<moab::EntityHandle> curve_j_verts;
+      result = get_meshset(*j, curve_j_verts);
+      assert(moab::MB_SUCCESS == result);
+      double j_curve_length = gen->length(curve_j_verts);
+
+      int i_id, j_id;
+      result = MBI()->tag_get_data(id_tag, &(*i), 1, &i_id);
+      assert(moab::MB_SUCCESS == result);
+      result = MBI()->tag_get_data(id_tag, &(*j), 1, &j_id);
+      assert(moab::MB_SUCCESS == result);
+      if (debug) {
+        std::cout << "curve i_id=" << i_id << " j_id=" << j_id
+                  << " leng0=" << curve_length << " leng1=" << j_curve_length << std::endl;
+      }
+
+      // reject curves with significantly different length (for efficiency)
+      if (facet_tol < abs(curve_length - j_curve_length))
+        continue;
+
+      // align j_curve to be the same as i_curve
+      bool reversed;
+      if (gen->dist_between_verts(curve_i_verts.front(), curve_j_verts.front()) >
+          gen->dist_between_verts(curve_i_verts.front(), curve_j_verts.back())) {
+        reverse(curve_j_verts.begin(), curve_j_verts.end());
+        reversed = true;
+      } else {
+        reversed = false;
+      }
+
+      // Reject curves if the average distance between them is greater than
+      // facet_tol.
+      double dist;
+      result = gen->dist_between_arcs(debug, curve_i_verts, curve_j_verts, dist);
+      assert(moab::MB_SUCCESS == result);
+      if (facet_tol < dist)
+        continue;
+
+      // THE CURVE WILL BE MERGED
+      if (debug) {
+        std::cout << "  merging curve " << j_id << " to curve " << i_id
+                  << ", dist_between_curves=" << dist << " cm" << std::endl;
+      }
+
+      // Merge the endpts of the curve to preserve topology. Merging (and deleting)
+      // the endpoints will also remove them from the KDtree so that the merged
+      // curve cannot be selected again. Update the curves when merging to avoid
+      // stale info.
+      if (curve_i_verts.front() != curve_j_verts.front()) {
+        result = zip->merge_verts(curve_i_verts.front(), curve_j_verts.front(),
+                                  curve_i_verts, curve_j_verts);
+        if (moab::MB_SUCCESS != result)
+          std::cout << result << std::endl;
```

Comment:
I would add a label to this `cout` to indicate what this result is related to, to help the user locate the error.  Both here and on line 667.